src/protocol/vm/state.rs

```diff
@@ -112,8 +112,8 @@ impl VMPoolState<PreCachedDB> {
     /// Ensures the pool supports the given capability
     fn ensure_capability(&self, capability: Capability) -> Result<(), SimulationError> {
         if !self.capabilities.contains(&capability) {
-            return Err(SimulationError::NotFound(format!(
-                "capability {:?}",
+            return Err(SimulationError::FatalError(format!(
+                "capability {:?} not found",
```

Comment:
I would word this as 'not supported' instead of 'not found'. To me 'not found' means it's not in the enum or something while 'not supported' means the pool doesn't support the capability